testing/integration/endpoint_security_test.go

```diff
@@ -169,3 +246,52 @@ func installElasticDefendPackage(t *testing.T, info *define.Info, policyID strin
 	require.NoError(t, err)
 	t.Logf("Endpoint package Policy Response:\n%+v", pkgResp)
 }
+
+func endpointIsHealthy(t *testing.T, ctx context.Context, agentClient client.Client) bool {
+	t.Helper()
+
+	state, err := agentClient.State(ctx)
+	if err != nil {
+		t.Logf("Error getting agent state: %s", err)
+		return false
+	}
+
+	if state.State != client.Healthy {
+		t.Logf("Agent is not Healthy\n%+v", state)
+		return false
+	}
+
+	foundEndpointInputUnit := false
+	foundEndpointOutputUnit := false
+	for _, comp := range state.Components {
+		isEndpointComponent := strings.Contains(comp.Name, "endpoint")
+		if comp.State != client.Healthy {
+			t.Logf("Component is not Healthy\n%+v", comp)
+			return false
+		}
+
+		for _, unit := range comp.Units {
+			if isEndpointComponent {
+				if unit.UnitType == client.UnitTypeInput {
+					foundEndpointInputUnit = true
+				}
+				if unit.UnitType == client.UnitTypeOutput {
+					foundEndpointOutputUnit = true
+				}
+			}
+
+			if unit.State != client.Healthy {
+				t.Logf("Unit is not Healthy\n%+v", unit)
```

Comment:
Nitpick: since we are testing the health of all components/units maybe we want to change the function name to something different? something like `agentAndEndpointHealthy` ?